src/libertem/job/masks.py

```diff
@@ -116,6 +136,16 @@ def _compute_masks(self):
         # If it is None, use sparse only if all masks are sparse
         # and set the use_sparse property accordingly
 
+        if not isinstance(self.mask_factories, Iterable):
```

Comment:
Can we somehow not repeat the conditionals here? I guess the only real difference is the last case, where we may have "mixed-sparsity" in the list of masks?